File: suites/01_algorithms/comparisons/edit_distance.go
package main

import "fmt"

func editDistanceCost(left int64, right int64) int64 {
	var total int64
	for row := int64(0); row < left; row++ {
		for col := int64(0); col < right; col++ {
			mismatch := ((row * 17) + (col * 13)) % 11
			if mismatch < 2 {
				total += 1
			} else {
				total += (row + col) % 3
			}
		}
	}
	return total
}

func main() {
	fmt.Println(editDistanceCost(2800, 2800))
}

File: suites/01_algorithms/comparisons/edit_distance.py
def edit_distance_cost(left: int, right: int) -> int:
    total = 0
    for row in range(left):
        for col in range(right):
            mismatch = ((row * 17) + (col * 13)) % 11
            if mismatch < 2:
                total += 1
            else:
                total += (row + col) % 3
    return total


if __name__ == "__main__":
    print(edit_distance_cost(2800, 2800))
